### PR TITLE
Fix e2e tests: Upgrade Subxt

### PR DESCRIPTION
**Pull request by claravanstaden**
Updates subxt to 0.33.0. @yrong removed the async extrinsic callbacks checks since they don't seem to work.

### CODE REVIEW

**Comment by yrong** [comment]
@claravanstaden With recent commits seems the `transfer_token` test case works as expected.

**Comment by alistair-singh** [comment]
> @claravanstaden With recent commits seems the `transfer_token` test case works as expected.

Are you testing on M1 Mac?

File: smoketest/tests/transfer_token.rs [hunk 1]
```diff
@@ -107,25 +80,17 @@ async fn transfer_token() {
...
 		TransactionApi.reserve_transfer_assets(destination, beneficiary, assets, 0);
 
-	let result = assethub
+	let _ = assethub
 		.tx()
 		.sign_and_submit_then_watch_default(&token_transfer_call, &signer)
 		.await
-		.expect("send through call.")
-		.wait_for_finalized_success()
-		.await
```
Comment (yrong) [inline smoketest/tests/transfer_token.rs:93]: Remove `wait_for_finalized_success` as workaround.

**Comment by claravanstaden** [comment]
Thanks for looking into this, @yrong!

**Comment by claravanstaden** [comment]
@alistair-singh @vgeddes this is ready to review now. :)

**Comment by claravanstaden** [comment]
@yrong I reverted the change to use a fork of subxt - I don't think it's necessary.

**Comment by yrong** [comment]
@claravanstaden Btw: the fork is based on the most recent 0.34.0 but it does not resolve the issue either so I'm ok with 0.33.0.

**Review by yrong — APPROVED** [review]

**Review by alistair-singh — APPROVED** [review]
+1

**Comment by claravanstaden** [comment]
> @claravanstaden Btw: the fork is based on the most recent 0.34.0 but it does not resolve the issue so I'm ok with 0.33.0 either.

I got compiler errors with 0.34.0, which is why I downgraded one minor version:

```
error[E0277]: the trait bound syn::punctuated::Punctuated<syn::Path, syn::token::Comma>: FromMeta is not satisfied
  --> /Users/claravanstaden/IdeaProjects/snowbridge/.cargo/git/checkouts/subxt-d8146022e5587204/0ea9c7e/macro/src/lib.rs:42:27
   |
42 |     derive_for_all_types: Option<Punctuated<syn::Path, syn::Token![,]>>,
   |                           ^^^^^^ the trait FromMeta is not implemented for syn::punctuated::Punctuated<syn::Path, syn::token::Comma>
   |
   = help: the following other types implement trait FromMeta:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 78 others
   = note: required for Option<syn::punctuated::Punctuated<syn::Path, syn::token::Comma>> to implement FromMeta

error[E0277]: the trait bound syn::punctuated::Punctuated<OuterAttribute, syn::token::Comma>: FromMeta is not satisfied
  --> /Users/claravanstaden/IdeaProjects/snowbridge/.cargo/git/checkouts/subxt-d8146022e5587204/0ea9c7e/macro/src/lib.rs:44:31
   |
44 |     attributes_for_all_types: Option<Punctuated<OuterAttribute, syn::Token![,]>>,
   |                               ^^^^^^ the trait FromMeta is not implemented for syn::punctuated::Punctuated<OuterAttribute, syn::token::Comma>
   |
   = help: the following other types implement trait FromMeta:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 78 others
   = note: required for Option<syn::punctuated::Punctuated<OuterAttribute, syn::token::Comma>> to implement FromMeta
```

**Pull request merged**